lib/logflare/backends.ex

```diff
@@ -1,51 +1,51 @@
 defmodule Logflare.Backends do
   @moduledoc false
-  alias Logflare.Backends.{
-    SourceBackend,
-    SourceDispatcher,
-    SourceRegistry,
-    SourceSup,
-    SourcesSup,
-    RecentLogs,
-    RecentLogsSup,
-    Adaptor.WebhookAdaptor
-  }
-
-  alias Logflare.{Buffers.MemoryBuffer, Source, LogEvent, Repo}
+  alias Logflare.Backends.Adaptor.WebhookAdaptor
+  alias Logflare.Backends.Adaptor.Postgres
+  alias Logflare.Backends.RecentLogs
+  alias Logflare.Backends.RecentLogsSup
+  alias Logflare.Backends.SourceBackend
+  alias Logflare.Backends.SourceDispatcher
+  alias Logflare.Backends.SourceRegistry
+  alias Logflare.Backends.SourcesSup
+  alias Logflare.Backends.SourceSup
+
+  alias Logflare.Buffers.MemoryBuffer
+  alias Logflare.LogEvent
+  alias Logflare.Repo
+  alias Logflare.Source
+
   import Ecto.Query
 
-  @adaptor_mapping %{
-    webhook: WebhookAdaptor
-  }
+  @adaptor_mapping %{webhook: WebhookAdaptor, postgres: Postgres}
 
   @doc """
   Lists `SourceBackend`s for a given source.
   """
   @spec list_source_backends(Source.t()) :: list(SourceBackend.t())
   def list_source_backends(%Source{id: id}) do
-    Repo.all(from sb in SourceBackend, where: sb.source_id == ^id)
-    |> Enum.map(fn sb ->
-      sb
-      |> typecast_config_string_map_to_atom_map()
-    end)
+    from(sb in SourceBackend, where: sb.source_id == ^id)
+    |> Repo.all()
+    |> Enum.map(fn sb -> typecast_config_string_map_to_atom_map(sb) end)
   end
 
   @doc """
   Creates a SourceBackend for a given source.
   """
   @spec create_source_backend(Source.t(), String.t(), map()) ::
           {:ok, SourceBackend.t()} | {:error, Ecto.Changeset.t()}
-  def create_source_backend(%Source{} = source, type, %{} = config) do
-    source
-    |> Ecto.build_assoc(:source_backends)
-    |> SourceBackend.changeset(%{config: config, type: type})
-    |> validate_config()
-    |> Repo.insert()
-    |> case do
+  def create_source_backend(%Source{} = source, type, config \\ %{}) do
+    source_backend =
+      source
+      |> Ecto.build_assoc(:source_backends)
+      |> SourceBackend.changeset(%{config: config, type: type})
+      |> validate_config()
+      |> Repo.insert()
+
+    case source_backend do
       {:ok, updated} ->
-        {:ok,
-         updated
-         |> typecast_config_string_map_to_atom_map()}
+        restart_source_sup(source)
```

Comment:
i don't think we should restart the entire source supervision tree here. it would unnecssarily restart unrelated source backend processes, as well as other processes under the SourcesSup.
It would instead be better to restart only the related source backend process (and its tree), via a kill or otherwise.

For example, say we ingest to postgres + bq + ch. updating the config for BQ would terminate processes for all 3 source backends.

Reply:
perhaps a `restart_source_backend_sup` function would help.

Reply:
This was a slight refactor for readability and this is something we can tackle in a later PR as an improvement of pipeline v2